frontend/src/components/contracts/ContractDetails.tsx

```diff
@@ -0,0 +1,155 @@
+import moment from "moment";
+
+import React from "react";
+
+import { Row, Col } from "react-bootstrap";
+
+import ContractsApi, * as C from "../../libraries/explorer-wamp/contracts";
+
+import CardCell from "../utils/CardCell";
+import TransactionLink from "../utils/TransactionLink";
+
+interface Props {
+  accountId: string;
+}
+
+interface State {
+  display: boolean;
+  locked: boolean;
+  contractInfo: C.ContractInfo;
+}
+
+export default class extends React.Component<Props, State> {
+  state: State = {
+    display: false,
+    locked: true,
```

Comment:
It is not a safe default. There is no good default, so it is better to start it as undefined (use question mark on the field definition in the State interface).